### DISCUSSION TITLE
Feature request - Email address for remote drops of recipe URLs?

### DISCUSSION THREAD

**Discussion by ttownsend** [nextcloud/cookbook#1295] [Ideas and discussions]
Hi folks,

First off - GREAT app.  This is exactly what I need to keep myself organized around recipes I find and want to try.

On THAT idea, could I request a feature - an email address (configured in Nextcloud) where I could send recipe URLs and your app would do its magic and import them here?

Many thanks and keep up the great work!

**Comment by christianlupus** [comment]
I am not aware of an option to read an email within Nextcloud except for the mail app. This might be implemented but I want to be sure, that this is in fact the best solution for you. Therefore, let me ask you a few questions.

Are you capable or not to reach the web interface of your NC installation when you would otherwise send an email back to yourself? Would it be feasible to install a bookmark in the browser? Are you using the browser alone and no other person is able to access it?

I am more thinking towards #431. That might be a simpler solution instead of implementing some IMAP functionality from scratch.

**Reply by ttownsend** [reply]
I'm not a coder, just a user - I was considering the use case where 1) I'm on a recipe website and find one that like, 2) using either the website's Share function or my own email client, I could send the URL of a found recipe to a unique email address for your app, and 3) your app would parse the recipe from the URL and store it.

**Reply by MarcelRobitaille** [reply]
For your described use case, wouldn't it be easier if the nextcloud cookbook app showed up as an option to share to? That workflow seems like it would be more convenient for you and way easier to implement than adding email to this app.

**Reply by christianlupus** [reply]
@MarcelRobitaille you are talking of an android app to allow for sharing, right? I have never  seen an option to share with a web page... But please correct me here. Or are you thinking towards a PWA?

**Reply by MarcelRobitaille** [reply]
@christianlupus Yes, I'm thinking this should be implemented in the cookbook android apps.

**Reply by ttownsend** [reply]
> For your described use case, wouldn't it be easier if the nextcloud cookbook app showed up as an option to share to? That workflow seems like it would be more convenient for you and way easier to implement than adding email to this app.

Yes, an option to "Share to cookbook" would be very useful (especially in Chrome where browser apps follow my profile across devices) however I think of similar use cases where I may be on some other phone/tablet/device where I do not have access to "Share to cookbook", so sending a URL to Nextcloud/recipe would allow your app to parse the received URL for a recipe when received from any source. 
Thoughts?

**Comment by MarcelRobitaille** [comment]
Maybe there could be an integration with Nextcloud Mail, so we wouldn't have to interact with IMAP directly.

But I agree. There might be a better workflow with a similar effect.

Is this on desktop or mobile? On mobile, a button to share a URL to the Cookbook app could be useful. But that would probably have to be implemented in one of the apps.

**Reply by christianlupus** [reply]
Sending emails seems possible to me. The reception is something different. This needs to use IMAP and a dedicated account.

As far as I know, the interconnection between apps is not discouraged but as least considered brittle. So, I would not bet that the mail app is capable of helping us much.

If we find something documented though, it might be well possible. The alternative is to use [the PHP IMAP extension](https://www.php.net/manual/de/book.imap.php).
